src/searchbar.cpp

```diff
@@ -164,40 +208,132 @@ void SearchBarLineEdit::updateCompletion()
         return;
     }
     m_token++;
-    auto suggestionWorker = new SuggestionListWorker(m_searchbarInput, m_token, this);
-    connect(suggestionWorker, &SuggestionListWorker::searchFinished, this,
-    [=] (const QStringList& suggestions, const QVector<QUrl>& urlList, int token) {
-        if (token != m_token) {
-            return;
+    fetchSuggestions(&SearchBarLineEdit::onInitialSuggestions);
+}
+
+void SearchBarLineEdit::fetchMoreSuggestions()
+{
+    /* TODO: Refactor suggestion worker to use single zim::SuggestionSearcher */
```

Comment:
This comment doesn't make much sense. A superficial review of the `SuggestionListWorker` will find a single instance of `zim::SuggestionSearcher`. Please explain the problem with this implementation so that an outsider (who didn't follow this PR) understands what exactly has to be addressed.

Reply:
Done